naming/dns_resolver_test.go

```diff
@@ -0,0 +1,304 @@
+package naming
+
+import (
+	"fmt"
+	"net"
+	"reflect"
+	"sync"
+	"testing"
+	"time"
+)
+
+func newUpdate(op Operation, addr string) *Update {
+	return &Update{
+		Op:   op,
+		Addr: addr,
+	}
+}
+
+func newUpdateWithMD(op Operation, addr, lb string) *Update {
+	return &Update{
+		Op:       op,
+		Addr:     addr,
+		Metadata: AddrMetadataGRPCLB{AddrType: GRPCLB, ServerName: lb},
+	}
+}
+
+func newSRVRR(target string, port uint16) *net.SRV {
+	return &net.SRV{
+		Target: target,
+		Port:   port,
+	}
+}
+
+var updateTc = []struct {
+	oldAddrs []string
+	newAddrs []string
+	want     []*Update
+}{
+	{
+		[]string{},
+		[]string{"1.0.0.1"},
+		[]*Update{newUpdate(Add, "1.0.0.1")},
+	},
+	{
+		[]string{"1.0.0.1"},
+		[]string{"1.0.0.1"},
+		[]*Update{},
+	},
+	{
+		[]string{"1.0.0.0"},
+		[]string{"1.0.0.1"},
+		[]*Update{newUpdate(Delete, "1.0.0.0"), newUpdate(Add, "1.0.0.1")},
+	},
+	{
+		[]string{"1.0.0.1"},
+		[]string{"1.0.0.0"},
+		[]*Update{newUpdate(Add, "1.0.0.0"), newUpdate(Delete, "1.0.0.1")},
+	},
+	{
+		[]string{"1.0.0.1"},
+		[]string{"1.0.0.1", "1.0.0.2", "1.0.0.3"},
+		[]*Update{newUpdate(Add, "1.0.0.2"), newUpdate(Add, "1.0.0.3")},
+	},
+	{
+		[]string{"1.0.0.1", "1.0.0.2", "1.0.0.3"},
+		[]string{"1.0.0.0"},
+		[]*Update{newUpdate(Add, "1.0.0.0"), newUpdate(Delete, "1.0.0.1"), newUpdate(Delete, "1.0.0.2"), newUpdate(Delete, "1.0.0.3")},
+	},
+	{
+		[]string{"1.0.0.1", "1.0.0.3", "1.0.0.5"},
+		[]string{"1.0.0.2", "1.0.0.3", "1.0.0.6"},
+		[]*Update{newUpdate(Delete, "1.0.0.1"), newUpdate(Add, "1.0.0.2"), newUpdate(Delete, "1.0.0.5"), newUpdate(Add, "1.0.0.6")},
+	},
+}
+
+func converToMap(u []*Update) map[string]*Update {
```

Comment:
conver**t**ToMap.  Or `updatesToMap` (or just `toMap` if this is the only thing like this).